### ISSUE TITLE
crashed: SQL logic error

### ISSUE THREAD

**Issue by CADbloke** [dhcgn/BeyondCompareSQLitePlugin#4]
Greetings

Thanks for writing this, I'm sure it's going to be really useful as soon as we fix it ...

I've tried this with no success in BC v 4.2.3 both with the latest release (very old) and also building it myself with the latest Nuget packages.

I got this ...

```
AssemblyResolve: BeyondCompareSQLitePlugin.Model.dll                                                                                                           
This program dumps a SQLite file to a text file.                                                                                                               
See wiki for instructions how to include this program in Beyond Compare 3 and 4.                                                                               
Link to wiki https://github.com/dhcgn/BeyondCompareSQLitePlugin/wiki                                                                                           
AssemblyResolve: System.Data.SQLite.dll                                                                                                                        
AssemblyResolve: EntityFramework.dll                                                                                                                           
Error ocurred: System.Data.SQLite.SQLiteException (0x80004005): SQL logic error                                                                                
near "-": syntax error                                                                                                                                         
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)                   
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()                                                                                                      
   at System.Data.SQLite.SQLiteCommand.GetStatement(Int32 index)                                                                                               
   at System.Data.SQLite.SQLiteDataReader.NextResult()                                                                                                         
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)                                                                     
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)                                                                                 
   at System.Data.SQLite.SQLiteCommand.ExecuteReader()                                                                                                         
   at BeyondCompareSQLitePlugin.Model.DbContext.GetColumnName(String table, SQLiteConnection dbConnection)                                                     
   at BeyondCompareSQLitePlugin.Model.DbContext.GetTableContent(List`1 tables, SQLiteConnection dbConnection)                                                  
   at BeyondCompareSQLitePlugin.Model.DbContext.GetTableContent(String source)                                                                                 
   at BeyondCompareSQLitePlugin.Program.MainInternal(String[] args) in C:\Codez\Github\BeyondCompareSQLitePlugin\BeyondCompareSQLitePlugin\Program.cs:line 39  
```

Even with an empty database (as well as a populated one) it threw the same error.

FYI in VS2017 my version didn't copy any of the `BeyondCompareSQLitePlugin.Model` project or the x86 or x64 folders to the `BeyondCompareSQLitePlugin` bin folder, which I thought weird because CopyLocal is set to true. You may want to try cloning it from Github to a bare repo and seeing how it progresses.

**Comment by dhcgn** [comment]
Could you please provide a sample sqlite file? 
I created an empty database and I get a different error (`System.InvalidOperationException`).

**Comment by CADbloke** [comment]
I can but I'd like to keep it private. Ping me by email & I'll send it to you. I'm my username at gmail.

I got that stack trace by running the plugin at the command line with parameters, fyi.

**Comment by CADbloke** [comment]
Thanks for fixing this. I tried to test it but ...

- Chrome failed the [zip download](https://github.com/dhcgn/BeyondCompareSQLitePlugin/releases/download/v1.2.1/BeyondCompareSQLitePlugin.zip) - "virus detected". MY AV is MS Security Essentials, Win7
- Install PS1 got the file permission messed up. I downloaded the PS1 & ran it in powershell.
![bs-sqlite-denied](https://user-images.githubusercontent.com/147329/36054967-5ad355d2-0e4d-11e8-9dd6-8579219c1514.png)
- uh, wait - that warning was because MS Security Essentials collared it
![bs-sqlite-trojan](https://user-images.githubusercontent.com/147329/36055050-ce5acea4-0e4d-11e8-803c-0883417f0d8e.png)
- uh, that didn't go so well. I'm assuming a false-positive there
- sorry I don't have time to do more testing, I'm knee-deep in other code atm but thanks again for the fix, the diff looks like it should work fine. I'll get back to you when I get time or hear more.

cheers
Ewen

**Comment by dhcgn** [comment]
I don't have the resource to buy a codes signing certificate to prevent these false positives.

I just tested the released executable, and it works with my test sqlite databse `Chinook_Sqlite_escape_needed.sqlite`.

The fix for your reported bug was to put single-quotes around each table and column name.

![fix3](https://user-images.githubusercontent.com/6566207/36060335-e4b27c0c-0e47-11e8-9a89-88201afd07bc.png)

I also tested this install script, it works in my setup.

```powershell
iwr -useb 'https://raw.githubusercontent.com/dhcgn/BeyondCompareSQLitePlugin/master/install.ps1' | iex;
```
